doc/api/async_hooks.md

```diff
@@ -613,15 +617,44 @@ class DBQuery extends AsyncResource {
 }
 ```
 
+#### `asyncResource.makeCallback(fn[, thisArg, ...args])`
+<!-- YAML
+added: REPLACEME
+-->
+
+* `fn` {Function} The function to call in the execution context of this async
+  resource.
+* `thisArg` {any} The receiver to be used for the function call.
+* `...args` {any} Optional arguments to pass to the function.
+
+Call the provided function with the provided arguments in the execution context
+of the async resource. This will establish the context, trigger the AsyncHooks
+before callbacks, call the function, trigger the AsyncHooks after callbacks, and
+then restore the original execution context.
+
 #### `asyncResource.emitBefore()`
+<!-- YAML
+deprecated: REPLACEME
+-->
+> Stability: 0 - Deprecated: Use [`asyncResource.makeCallback()`][] instead.
 
 * Returns: {undefined}
 
 Call all `before` callbacks to notify that a new asynchronous execution context
 is being entered. If nested calls to `emitBefore()` are made, the stack of
 `asyncId`s will be tracked and properly unwound.
 
+*Note*: `before` and `after` calls must be unwound in the same order that they
```

Comment:
Nit: can you just remove the `*Note*: ` prefix here? :-)

Reply:
Done.